pkgs/stdenv/generic/default.nix

```diff
@@ -184,15 +81,38 @@ let
       propagatedBuildInputs' = lib.chooseDevOutputs propagatedBuildInputs;
       propagatedNativeBuildInputs' = lib.chooseDevOutputs propagatedNativeBuildInputs;
 
-    in
-
-      # Throw an error if trying to evaluate an non-valid derivation
-      assert let v = checkValidity attrs;
-             in if !v.valid
-               then throwEvalHelp (removeAttrs v ["valid"])
-               else true;
+      pos' =
```

Comment:
While we're at it, can we make this just the default arg for `pos`, rather than using `, pos ? null` and `if pos != null`?